realm/src/androidTest/java/io/realm/RealmObjectTest.java

```diff
@@ -477,4 +477,35 @@ public void testThreadModelClass() {
         Thread thread = testRealm.createObject(Thread.class);
         testRealm.commitTransaction();
     }
+
+    public void testIsValidUnManagedObject() {
+        AllTypes allTypes = new AllTypes();
+        assertFalse(allTypes.isValid());
+    }
+
+    public void testIsValidClosedRealm() {
+        testRealm.beginTransaction();
+        AllTypes allTypes = testRealm.createObject(AllTypes.class);
+        assertTrue(allTypes.isValid());
+        testRealm.commitTransaction();
+        testRealm.close();
+        assertFalse(allTypes.isValid());
+    }
+
+    public void testIsValidDeletedObject() {
+        testRealm.beginTransaction();
+        AllTypes allTypes = testRealm.createObject(AllTypes.class);
+        assertTrue(allTypes.isValid());
+        testRealm.clear(AllTypes.class);
+        testRealm.commitTransaction();
+        assertFalse(allTypes.isValid());
+    }
+
+    public void testIsValidManagedObject() {
```

Comment:
I am not a big fan of unit tests that test different things as the naming become too vague, and although the unit tests look kinda similar they test different code paths. So I will probably keep them as is, unless the others also object :)